dbms/src/Common/Decimal.h

```diff
@@ -108,66 +108,40 @@ struct IntPrec<Int256>
 
 struct PlusDecimalInferer
 {
-    static std::tuple<PrecType, ScaleType> infer(PrecType left_prec, ScaleType left_scale, PrecType right_prec, ScaleType right_scale)
-    {
-        ScaleType result_scale = std::max(left_scale, right_scale);
-        PrecType result_int = std::max(left_prec - left_scale, right_prec - right_scale);
-        PrecType result_prec = std::min(result_scale + result_int + 1, decimal_max_prec);
-        return {result_prec, result_scale};
-    }
+    static std::tuple<PrecType, ScaleType> infer(PrecType left_prec, ScaleType left_scale, PrecType right_prec, ScaleType right_scale);
```

Comment:
This pr moves lost of implementations of inline functions into cpp files, so they are never inlineable, which could hurt the performance, is that what you expect?

Reply:
In https://github.com/pingcap/tiflash/pull/4890 and https://github.com/pingcap/tiflash/pull/4924, we have enabled LTO  in release build process. Compiler will inline those short methods automatically.

For example: use `objdump -d`, there will be only entry to function in code section. There is no other function call towards `PlusDecimalInferer::infer` by instruction `callq`.
```
➜  release-centos7-llvm git:(pch) ✗ ./tiflash/tiflash version
TiFlash
Release Version: v6.2.0-alpha-30-gf2a406a
Edition:         Community
Git Commit Hash: f2a406aa93f06e98e1868597f0ff25ae87ed40c5
Git Branch:      pch
UTC Build Time:  2022-06-05 07:30:43
Enable Features: jemalloc avx avx512 unwind thinlto
Profile:         RELWITHDEBINFO

Raft Proxy
Git Commit Hash:   ca2f51f94e55bdd23749dcc02ab4afb94eeb5ae5
Git Commit Branch: HEAD
UTC Build Time:    2022-06-05 07:33:25
Rust Version:      rustc 1.60.0-nightly (1e12aef3f 2022-02-13)
Storage Engine:    tiflash
Prometheus Prefix: tiflash_proxy_
Profile:           release

➜  release-centos7-llvm git:(pch) ✗ objdump -d ./tiflash/tiflash | grep 'PlusDecimalInferer'
0000000001ce43c0 <_ZN2DB18PlusDecimalInferer5inferEjjjj>:```

➜  release-centos7-llvm git:(pch) ✗ objdump -d ./tiflash/tiflash | grep '1ce43c0'           
0000000001ce43c0 <_ZN2DB18PlusDecimalInferer5inferEjjjj>:
 1ce43c0:       55                      push   %rbp

Reply:
Although I think there is no risk, but... follow reviewer's idea.